code/datums/diseases/_disease.dm

```diff
@@ -65,15 +65,17 @@
 
 	if(!cure)
 		if(prob(stage_prob))
-			stage = min(stage + 1,max_stages)
+			update_stage(min(stage + 1,max_stages), src)
 	else
 		if(prob(cure_chance))
-			stage = max(stage - 1, 1)
+			update_stage(max(stage - 1, 1), src)
 
 	if(disease_flags & CURABLE)
 		if(cure && prob(cure_chance))
 			cure()
 
+/datum/disease/proc/update_stage(new_stage, datum/disease/A)
```

Comment:
A is just src, unnecessary

Reply:
Thanks it should no longer pass it as a parameter